Mapbox/MapboxMapsLocation/Pucks/LocationPuckManager.swift

```diff
@@ -20,15 +20,18 @@ internal enum PuckStyle {
 // MARK: PuckBackend
 /// This enum represents the different backends that can be used for Pucks
 public enum LocationPuck: Equatable {
-    case puck2D(customize: ((inout LocationIndicatorLayerViewModel) -> Void)? = nil) // Backed by `LocationIndicatorLayer`. Implement customize block to granularly modify the puck's styling.
-    case puck3D(customize: ((inout PuckModelLayerViewModel) -> Void))// Backed by `ModelLayer`. Implement customize block to granularly modify the puck's styling.
+    /// Backed by `LocationIndicatorLayer`. Optionally provide `LocationIndicatorViewModel` to granularly modify the puck's styling.
+    case puck2D(LocationIndicatorLayerViewModel? = nil)
+
+    /// Backed by `ModelLayer`.
+    case puck3D(PuckModelLayerViewModel)
 
     public static func == (lhs: LocationPuck, rhs: LocationPuck) -> Bool {
         switch (lhs, rhs) {
-        case (.puck2D(_), .puck2D(_)):
-            return true
-        case (.puck3D(_), .puck3D(_)):
-            return true
+        case (.puck2D(let lhsVM), .puck2D(let rhsVM)):
+            return lhsVM == rhsVM
+        case (.puck3D(let lhsVM), .puck3D(let rhsVM)):
+            return lhsVM == rhsVM
         default:
             return false
         }
```

Comment:
Since both associated value types are `Equatable`, can this be synthesized?

Reply:
Oh seems like it can! This is definitely new though. Can get rid of the whole thing now! 👍🏾